### PR TITLE
v1-of-profile-for-student

### PR DESCRIPTION
**Pull request by pr0meth4us**

### CODE REVIEW

File: backend/models/student.py
Comment (pr0meth4us) [inline backend/models/student.py:1]: i think this is not the best practice. what it does is it creates a new database to store the info just for student profile, and this way we're gonna need another one for instructors.

File: backend/models/student.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from bson import ObjectId
+
+class Student:
+    def __init__(self, student_id, name, email, bio, courses_enrolled, profile_image):
+        self.student_id = student_id
+        self.name = name
+        self.email = email
+        self.bio = bio
+        self.courses_enrolled = courses_enrolled
+        self.profile_image = profile_image
```
Comment (pr0meth4us) [inline backend/models/student.py:10]: `
```suggestion
        self.profile_image = profile_image
```

File: backend/controllers/auth_controller.py [hunk 2]
```diff
@@ -32,11 +34,29 @@ def register(data):
...
+                "student_id": data.get("student_id"),  # Ensure this data is provided in the request
+                "name": data['name'],
+                "email": email,
+                "bio": data.get("bio", ""),  # Optional bio
+                "courses_enrolled": [],  # Initialize as empty, can be updated later
+                "profile_image": data.get("profile_image", "")  # Optional image path
+            }
+
+            # Create student profile using the StudentController
+            student_response = StudentController.create_student_profile(student_data)
```
Comment (pr0meth4us) [inline backend/controllers/auth_controller.py:49]: i think this is not the best practice. what it does is it creates a new database to store the info just for student profile, and this way we're gonna need another one for instructors.

File: backend/models/student.py
Comment (pr0meth4us) [inline backend/models/student.py:1]: i suggest this should be deleted then you can put the attributes    
 self.bio = bio
        self.courses_enrolled = courses_enrolled
        self.profile_image = profile_image
        into the user model instead

**Review by pr0meth4us — COMMENTED** [review]
You also need a middleware so it allow users to only edit their own profile. and for the profile image you need an image upload function and a new noSql database to upload the images to

**Review by VathanakBotraDos — APPROVED** [review]

File: backend/controllers/student_controller.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        image = request.files['image']
+
+        # Validate the image (you may want to add more checks for file types, size, etc.)
+        if image.filename == '':
+            return jsonify({"message": "No selected file"}), 400
+
+        # Save the image (update this path as needed for your project)
+        upload_folder = 'path/to/upload/folder'  # Define the upload folder
+        image_path = os.path.join(upload_folder, image.filename)
```
Comment (pr0meth4us) [inline backend/controllers/student_controller.py:68]: this actually upload the img into a folder on ur laptop, which is ok. but better yet we can make it upload to a cloud databse

**Review by VathanakBotraDos — APPROVED** [review]

**Pull request merged**